.github/workflows/publish.yml

```diff
@@ -1,32 +1,32 @@
-name: Publish to npm
+name: Release
+
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - main
+      - 'pre-release-beta/**'
+      - 'beta/**'
+      - 'prototype-beta/'
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
   publish-npm:
+    name: Release
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version: 16
           registry-url: https://registry.npmjs.org/
-      - name: Check release validity
-        run: sh .github/scripts/check-release.sh
-      - name: Check tag format
-        run: sh .github/scripts/check-tag-format.sh "${{ github.event.release.prerelease }}"
```

Comment:
No need to check the tag format anymore?

Reply:
I'll maybe add it in the PR that tries the different beta types if it is relevant